src/pages/instances/forms/InstanceFormMenu.tsx

```diff
@@ -54,7 +62,19 @@ const InstanceFormMenu: FC<Props> = ({
               <MenuItem label={CLOUD_INIT} {...menuItemProps} />
             </ul>
           </li>
-          <MenuItem label={YAML_CONFIGURATION} {...menuItemProps} />
+          {isConfigDisabled ? (
+            <li className="p-side-navigation__item">
+              <button
+                className="p-side-navigation__link p-button--base"
+                disabled={true}
+                title="Please select an image before adding custom configuration"
+              >
+                {YAML_CONFIGURATION}
+              </button>
```

Comment:
Why aren't we using react-component's `Button` here? (and also @ L39 above and in `ProfileFormMenu` @ L37)

Also, please note that `p-button--base` has `text-align: center;`. You might want to override that:

![image](https://user-images.githubusercontent.com/56583786/225763790-353b0163-67ff-4f2c-a155-71a7c877119c.png)
(disabled, no image selected)

vs

![image](https://user-images.githubusercontent.com/56583786/225763867-a8cfba9d-bd07-4a85-86b6-d63a4e59f23d.png)
(enabled, after image selection)

Reply:
Good catch, adjusted the usages to `Button` and fixed the alignment / width.